README.md

```diff
@@ -1 +1,177 @@
+
+For example, to repackage your `dev` and `release` directories automatically on a code change,
+
+```shell
+dojo-tsc-watcher -p tsconfig.json tsconfig.esm.json -- npm run package
+```
+
+By default, the watcher will *not* run the target script if compilation fails on one of the tsc processes. If you want run the target script regardless of if the compilation succeeded or failed, you can pass the `—force` flag.
+
+#### Linting
+
+This package also includes a base set of tslint rules you can use. Update your `tslint.json` to include,
+
+```json
+{
+  "extends": "./node_modules/@dojo/scripts/tslint/base.json"
+}
+```
+
+You can now lint your project with,
+
+```shell
+$ tslint -p .
+```
+
+### Packaging
+
+The provided `dojo-package` script will take all of the directories in the `dist` directory (`umd`, `cjs`, and `esm`) and merge them together to create  `dev` and `release` directories. When you run tests, they are run from the `dev` directory. Production, or release, code is stored in the `release` directory. The release directory will contain the files from the `dev/src` directory as well as a modified `package.json` from your project root.
+
+Once in this format, you can easily create a `.tar.gz` of your package with `npm pack dist/release`.
+
+### Testing
+
+Projects can extend the provided Intern configs and avoid boilerplate configuration. To use the intern config, create an `intern.json` in your project with the following,
+
+```json
+{
+  "extends": "./node_modules/@dojo/scripts/intern/base.json",
+  "capibilities": {
+    "name": "@dojo/your-project"
+  }
+}
+```
+
+In this file, you can add further configuration to override the base config (for example, custom loader configuration).
+
+Now, with a regular `intern.json` you can run Intern from the command line to run your tests.
+
+```shell
+# run node unit tests
+$ npx intern
+
+# run browser unit tests
+$ npx intern config=intern.json@local
+```
+
+See the [Intern docs](https://github.com/theintern/intern/blob/master/docs/running.md) for more options on running Intern.
+
+### Releasing
+
+Several scripts are provided to ease the release process.
+
+#### Can Publish Checks
+
+To check if the user is allowed to publish, run the `dojo-can-publish-check` script. The script will fail with a `1` exit code if the user cannot publish.
+
+#### Clean Repo Checks
+
+A safe release is a clean release. To check if there are no uncommitted changes, and the user is `master`, run the`dojo-repo-is-clean-check` script. The script will fail with a `1` exit code if the repo is dirty.
+
+#### Release
+
+The `dojo-release` script can release a dojo package. The `dist/release` directory is what gets released. The script takes a number of arguments:
+
+| Parameter  | Description                              |
+| ---------- | ---------------------------------------- |
+| `—release` | The version to release                   |
```

Comment:
Perhaps we need to add an `--initial` flag that adds `--access public` to the publish command or maybe we can just include that all the time (for every publish).

Reply:
Also for `--initial` we'll need to skip the check against the existing maintainers (because there won't be any)

Reply:
Added an `--initial` (default false) parameter to the release script. The can-publish checks are actually done by the `dojo-can-publish-check` script. For the initial release people can just not run this script?